packages/components/src/Notification/subcomponents/GenericNotification/GenericNotification.tsx

```diff
@@ -0,0 +1,198 @@
+import React from "react"
+import classnames from "classnames"
+import { Heading, HeadingProps } from "@kaizen/typography"
+import { NotificationType } from "~components/Notification/types"
+import { CautionIcon } from "~icons/CautionIcon"
+import { CloseIcon } from "~icons/CloseIcon"
+import { ExclamationIcon } from "~icons/ExclamationIcon"
+import { InformationIcon } from "~icons/InformationIcon"
+import { SecurityTipIcon } from "~icons/SecurityTipIcon"
+import { SuccessIcon } from "~icons/SuccessIcon"
+import { OverrideClassName } from "~types/OverrideClassName"
+import styles from "./GenericNotification.module.scss"
+
+export type GenericNotificationProps = OverrideClassName<{
+  type: NotificationType
+  style: "global" | "inline" | "toast"
+  children?: React.ReactNode
+  title?: string
+  persistent: boolean
+  onHide?: () => void
+  noBottomMargin?: boolean
+  forceMultiline?: boolean
+  headingProps?: HeadingProps
+}>
+
+type State = {
+  hidden: boolean
+  removed: boolean
+}
+
+const renderIcon = (type: NotificationType): JSX.Element => {
+  switch (type) {
+    case "positive":
+      return <SuccessIcon role="presentation" inheritSize />
+    case "negative":
+      return <ExclamationIcon role="presentation" inheritSize />
+    case "cautionary":
+      return <CautionIcon role="presentation" inheritSize />
+    case "informative":
+      return <InformationIcon role="presentation" inheritSize />
+    case "security":
+      return <SecurityTipIcon role="presentation" inheritSize />
+    default:
+      return <InformationIcon role="presentation" inheritSize />
+  }
+}
+
+class GenericNotification extends React.Component<
```

Comment:
Create a new top level folder for this, something which I'm also doing for `Button`